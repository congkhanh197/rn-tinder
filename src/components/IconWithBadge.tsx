import React from 'react';
import {View, Text} from 'react-native';
import Ionicons from 'react-native-vector-icons/Ionicons';

interface IProps {
  name: string;
  badgeCount: number;
  color: string;
  size: number;
}
const IconWithBadge: React.FC<IProps> = props => {
  const {name, badgeCount, color, size} = props;
  return (
    <View style={{width: 24, height: 24, margin: 5}}>
      <Ionicons name={name} size={size} color={color} />
      {badgeCount > 0 && (
        <View
          style={{
            position: 'absolute',
            right: -6,
            top: -3,
            backgroundColor: 'red',
            borderRadius: 6,
            width: 12,
            height: 12,
            justifyContent: 'center',
            alignItems: 'center',
          }}>
          <Text style={{color: 'white', fontSize: 10, fontWeight: 'bold'}}>
            {badgeCount}
          </Text>
        </View>
      )}
    </View>
  );
};
export default IconWithBadge;
