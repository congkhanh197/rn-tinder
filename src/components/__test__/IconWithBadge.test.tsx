import React from 'react';
import IconWithBadge, { IProps } from '../IconWithBadge';
import renderer from 'react-test-renderer';
describe('Test Card Item', () => {
  test('should render', () => {
    const props: IProps = {
        name: "testName",
        badgeCount: 1,
        color: "blue",
        size: 23
    }
    const tree = renderer.create(<IconWithBadge {...props}/>);
    expect(tree).toMatchSnapshot();
  });
});
