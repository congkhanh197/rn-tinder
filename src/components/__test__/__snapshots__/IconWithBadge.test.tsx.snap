// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Item should render 1`] = `
<View
  style={
    Object {
      "height": 24,
      "margin": 5,
      "width": 24,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "blue",
          "fontSize": 23,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 6,
        "height": 12,
        "justifyContent": "center",
        "position": "absolute",
        "right": -6,
        "top": -3,
        "width": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 10,
          "fontWeight": "bold",
        }
      }
    >
      1
    </Text>
  </View>
</View>
`;
